### DISCUSSION TITLE
Take advantage of YAML-based configuration

### DISCUSSION THREAD

**Discussion by lindamejia** [open-sdg/open-sdg#1913] [Q&A]
I'm in the process of updating the Los Angeles SDG site utilizing these instructions: https://open-sdg.readthedocs.io/en/latest/upgrades/upgrading-1-0-0/
In order to take advantage of the YAML- based configuration do all existing .MD file have to be converted to .YML?

**Comment by brockfanning** [comment]
@lindamejia You can continue to use .MD files for the indicator configuration, if you prefer. The "YAML-based configuration" in that documentation is referring to a `config_data.yml` file that can be used to control the behavior of the data build.

There are a two caveats though:

1. You may be interested in switching from .MD to .YML for indicator configuration, eventually, to take advantage of the "indicator configuration forms". I believe these were added in version 1.4.0, so you would not be able to use them until you get to that version. That feature provides a form for indicator configuration that automatically generates a .YML file. So it might be more convenient if you have switched to .YML by then. Using this feature is not a requirement, though.
2. That documentation mentions "emulating the data starter" and links directly to the current version of the `config_data.yml` file. However that file has changed a lot since version 1.0.0 was released. That documentation really should have linked to a specific commit, to show what the file was like at that specific time. Here is a snapshot of that `config_data.yml` file at the time that 1.0.0 was released: https://github.com/open-sdg/open-sdg-data-starter/blob/07d881fe11cc2cb5e3cf060fec49cc752d6d8227/config_data.yml.
Notice that in the `inputs` section, on line 44, it specifies an input called `InputYamlMdMeta`. This is what allows you to continue using .MD files for indicator configuration. If you were to switch to .YML for indicator configuration, you would need to switch that to `InputYamlMeta` (ie, remove the `Md`).

Hope that makes sense, and feel free to follow up here if there are any more issues with the upgrade.

**Reply by lindamejia** [reply]
@brockfanning Thank you so much for your quick response. I just want to confirm my assumption, in order to get this site up to date, I will need to update it from the current version to version 1.0.0 then to 1.2.0 then 1.3.0 and so on. Is this correct? Or in general, what do you recommend would be the best way to go about updating this site? 
Also, do you happen to still have the old documentation? (It's been a while since I looked at this so I'm trying to retrace my steps).

**Reply by brockfanning** [reply]
@lindamejia We try our best now to introduce breaking changes in minor releases, so in theory you could get to the latest release (2.1.0) in four steps:
1. Upgrade to 1.0.0
2. Upgrade from 1.0.0 to 1.8.1
3. Upgrade from 1.8.1 to 2.0.0
4. Upgrade from 2.0.0 to 2.1.0

The difficulty of the steps depends on how many (and which) files are being overridden in your implementation.

For old documentation, see the drop-down in the lower right - you should be able switch between older versions of the documentation with that.
